### PR TITLE
Max score

### PR DESCRIPTION
**Pull request by tsivert**
Jira issue: https://h5ptechnology.atlassian.net/browse/HFP-1808

Added support for calculating the maximum score of the content type.

### CODE REVIEW

File: presave.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+var H5PPresave = H5PPresave || {};
+
+H5PPresave['H5P.MarkTheWords'] = function (content, finished) {
+    var score = 0;
+    if( content.hasOwnProperty('textField') === true ){
+        var pattern = /\*[^\*\s]+\*/g;
+        score = content.textField.match(pattern || []).length;
```
Comment (thomasmars) [inline presave.js]: This does not account for multi-word answers that users may have made, e.g. *multi word answer*. 
It also does not account for words that has escaped asterisks within them, e.g.: *cor**rect*

Comment (thomasmars) [reply presave.js]: The max score logic should optimally be exactly the same as the score parser that is used within the view.

Comment (tsivert) [reply presave.js]: Ok. In the Important instructions in the editor it says that only words may be marked as correct, not phrases. That's why I didn't add support for multiple word answers.
I'll fix it.

**Comment by fnoks** [comment]
@tsivert  FYI: I have done some changes in presave.js across all content types:
1. renamed isContentInValid -> isContentInvalid (You can't be inside a valid ;)
2. removed the check for a undefined finished function. This function should always be invoked with a callback. In case there's a future bug where the callback is not provided, it is easier to catch it if we see an exception.
3. Minor code cleanup (for things found by eslint)

If you disagree with any of these (especially #2), please let me know.

**Comment by tsivert** [comment]
@fnoks I have absolutely no problems with your changes. :)

**Pull request merged**